source/grid/grid_generator.cc

```diff
@@ -4991,6 +4991,281 @@ namespace GridGenerator
 
 
 
+  template <int dim>
+  void
+  non_standard_orientation_mesh(Triangulation<dim> &tria,
+                                const bool          face_orientation,
+                                const bool          face_flip,
+                                const bool          face_rotation,
+                                const bool          manipulate_left_cube)
+  {
+    if (dim < 3)
+      AssertThrow(false, ExcMessage("This mesh is only implemented in 3D."));
```

Comment:
Do you have plans of implementing the 2d part as well?

Reply:
I implemented another function doing a similar thing for 2d meshes. However the interface is different so I renamed the functions to <code>non_standard_orientation_mesh_2D</code> and <code>non_standard_orientation_mesh_3D</code>.

When checking the respective 2d orientations in a test it seems like that the mesh generator takes care of that (see test results): the flags for <code>cell->face_orientation(face_index)</code> <code>cell->line_orientation(line_index)</code> are always true, i.e., the faces/lines are in standard orientation. I guess that is ok since planar 2d meshes can always be oriented consistently? In 3d the situation is different as can also be seen from the tests. 

This also relates to my other PR #11414. The orientation issue does not arise in 2d.

Reply:
You're correct that meshes in 2d can always be ordered so that face orientations match. We wrote a whole paper about it: https://www.math.colostate.edu/~bangerth/publications/2015-graph.pdf One of the more fun papers I've worked on in my career :-)

But I didn't think that we do that by default. Are you saying that when you describe a mis-ordered mesh to `Triangulation::create_triangulation()`, that you receive a correctly ordered one?